dicom/matRad_importDicom.m

```diff
@@ -161,3 +162,21 @@
         resultGUI.w = [resultGUI.w; [stf(i).ray.weight]'];
     end
 end
+
+
+%% save ct, cst, pln, dose
+matRadFileName = [files.ct{1,3} '.mat']; % use default from dicom
+[FileName,PathName] = uiputfile('*','Save as...',matRadFileName);
+if ischar(FileName)
+    % delete unnecessary variables
+    switch env
+    case 'MATLAB'
+        clearvars -except ct cst pln stf resultGUI FileName PathName;
+        save([PathName, FileName], '-regexp', '^(?!(FileName|PathName)$).','-v7');
+    case 'OCTAVE' 
+        clear -x ct cst pln stf resultGUI FileName PathName;
+        save([PathName, FileName],'-v6');
```

Comment:
Is this so smart? If  a user runs the import and has some important work in the workspace everything will be cleared. I think this can be avoided by doing a sequential save with everything that could be loaded. You start with the CT in line 75. Then you append structures using the 'append' argument with the save command whenever a new structure is created.

In order to avoid repeated code this saving can be factored out in a little helper function matRad_save(filename, variablename, appendFlag)

https://de.mathworks.com/help/matlab/ref/save.html#d124e1343226